includes/notes/class-wc-payments-notes-additional-payment-methods.php

```diff
@@ -43,6 +50,13 @@ public static function get_note() {
 			return;
 		}
 
+		// if the user hasn't connected their account (or the account got disconnected) do not add the note.
+		if ( self::$account instanceof WC_Payments_Account ) {
+			if ( false === self::$account->is_stripe_connected() ) {
```

Comment:
That's the intent, I'm not 100% sure if `get_note` can be called with `$account` not being set (since it's a static method), so I just let the note get added if it's not set